Vassal conversion code improvement & Monster Hunter changes

JohnFulpWillard opened the pull request:
## About The Pull Request

Moves Bloodsucker & Vassal conversion helpers and checks into their own file instead of being thrown in with the rulesets
Reworks them to be easier to read, and fixes Vassalizing people whose masters broke the Masquerade.
Makes better var names for all the procs and documents all of them
Also removes AmInvalidAntag

### Monster Hunter changes

Monster hunters are now giving antag datums based off of a weight, with Blood cultists and Wizard added to the mix.
Monster Hunters now will also no longer spawn people who can't be owned as antags, nor will they count them towards their required list of monsters.

## Why It's Good For The Game

The code is less more resiliant to breaking and is more readable for contributors.
Also fixes a mechanic that was added a while ago, if it ever worked.

## Changelog

:cl:
fix: Deactivating a Power will no longer make you use the power when you next click.
fix: Bloodsuckers can now steal Vassals of those who broke the Masquerade.
fix: You can no longer turn invalid mobs (like Plasmamen) into special Vassals.
fix: Monster Hunters will no longer create Plasmamen Bloodsuckers and instantly kill them,
/:cl:

JohnFulpWillard committed:
Improves vassal conversion

JohnFulpWillard committed:
better comment

JohnFulpWillard committed:
i hate tgstation.dme

JohnFulpWillard committed:
Uses of make_bloodsucker are more sane

JohnFulpWillard committed:
Unconvertable & ValidAntag fix

JohnFulpWillard committed:
Check can_be_owned on Bloodsuckers

JohnFulpWillard committed:
code simplification

JohnFulpWillard committed:
minor fix

SgtHunk commented on fulp_modules/features/antagonists/bloodsuckers/code/rulesets/monsterhunter_event.dm:
```diff
@@ -1,3 +1,13 @@
+///List of antagonists that are considered 'Monsters' and their chance of being selected.
+GLOBAL_LIST_INIT(monster_antagonist_types, list(
+	/datum/antagonist/bloodsucker = 50,
+	/datum/antagonist/heretic = 50,
+	/datum/antagonist/changeling = 20,
+	/datum/antagonist/cult = 3,
+	/datum/antagonist/wizard/apprentice/imposter = 1,
+))
```
mfw values don't add to 100

remove32 replied:
relative weights woah woah woah

JohnFulpWillard replied:
They don't have to

SgtHunk replied:
would be a lot cooler if they did

JohnFulpWillard replied:
It doesn't matter it's just weight

OneAsianTortoise commented:
Hey man it is FFF

SmoSmoSmoSmok commented on fulp_modules/features/antagonists/bloodsuckers/code/rulesets/monsterhunter_event.dm:
```diff
@@ -42,31 +51,26 @@
 		var/mob/living/monster = pick(living_players)
+		var/datum/antagonist/profession = pick_weight(GLOB.monster_antagonist_types)
+		if(!monster.mind.add_antag_datum(profession))
+			continue
```
this iteration will be counted in the loop, creating 1 less monster than we need everytime the monster can't be given the antag

JohnFulpWillard replied:
it adds to the amount to add if it fails

JohnFulpWillard committed:
Merge branch 'master' into vassal-conversion

JohnFulpWillard committed:
Update monsterhunter_event.dm

JohnFulpWillard committed:
Update monsterhunter_event.dm

JohnFulpWillard committed:
Bloodsuckers hopefully work lol

Bloodsuckers now actually use blood when Vassalizing
Non-antags no longer have to manually accept vassalization
People resisting vassalization can resist it easier now and are on a timer
Persuasion rack has a better description
You can turn people into special vassals now
Targeted powers can now be disabled without making you still perform the power on your next click

JohnFulpWillard committed:
Merge branch 'vassal-conversion' of https://github.com/JohnFulpWillard/fulpstation into vassal-conversion

JohnFulpWillard committed:
Moves files around

Moves Bloodsucker datum to the Bloodsucker folder
Moves all antagonist rulesets into their own 'rulesets' folder

JohnFulpWillard committed:
Moves more files around

JohnFulpWillard committed:
unconvertable

JohnFulpWillard committed:
Update monsterhunter_event.dm

JohnFulpWillard committed:
Merge branch 'vassal-conversion' of https://github.com/JohnFulpWillard/fulpstation into vassal-conversion

JohnFulpWillard committed:
fucked up bruh

Carlito-byte commented:
bad code, made it so if you get converted as a vassal you can't even see the hud mark of your master. relogging doesn't fix it either.

JohnFulpWillard committed:
Merge branch 'master' into vassal-conversion

Commit:
mmmm (#56)

JohnFulpWillard commented:
Bloodsucker POV
![image](https://user-images.githubusercontent.com/53777086/216821708-f11040c4-9f7b-4129-bf42-d1247b8e36b0.png)
![image](https://user-images.githubusercontent.com/53777086/216821712-b0c64ba1-0a95-44a8-b862-9a2b0400517f.png)

Vassal POV
![image](https://user-images.githubusercontent.com/53777086/216821727-31e21252-6d5d-48c2-b6d8-9e230154eb33.png)
![image](https://user-images.githubusercontent.com/53777086/216821728-49eb1d3b-8467-4617-916b-75deb2512754.png)
